google/api_core/path_template.py

```diff
@@ -308,9 +308,27 @@ def transcode(http_options, message=None, **request_kwargs):
             else:
                 try:
                     if message:
-                        request["body"] = getattr(leftovers, body)
-                        delete_field(leftovers, body)
+                        try:
+                            request["body"] = getattr(leftovers, body)
+                            delete_field(leftovers, body)
+                        except AttributeError as e:
+                            # gapic-generator-python appends underscores to field names
+                            # that collide with python keywords.
+                            # `_` is stripped away as it is not possible to
+                            # natively define a field with a trailing underscore in protobuf.
+                            # See related issue
+                            # https://github.com/googleapis/python-api-core/issues/227
+                            if hasattr(leftovers, body + "_"):
```

Comment:
If we know that the trailing underscore will fail, why do we have to get an exception before we deal with this? Why not take processing similar to that in lines 327 _et seq_ (stripping `"_"` from `leftovers`) and put it above, and possibly get rid of the `try` block? (Side effect would be also stripping underscores from the other fields in `leftover`, which the code here doesn't do, but I guess we want, so the query params don't have underscores?)

Reply:
I haven't found a way to strip the trailing underscore from a protobuf message. We need to use the correct suffix to access the field. I've proposed a new version without the `try` block. PTAL